lib/temporal/metadata.rb

```diff
@@ -33,12 +35,13 @@ def generate_workflow_task_metadata(task, namespace)
           attempt: task.attempt,
           workflow_run_id: task.workflow_execution.run_id,
           workflow_id: task.workflow_execution.workflow_id,
-          workflow_name: task.workflow_type.name
+          workflow_name: task.workflow_type.name,
+          query_type: task.query&.query_type,
+          query_args: from_query_payloads(task.query&.query_args)
```

Comment:
I wonder what's the use-case for exposing query via metadata?

Reply:
Also if we do this we'll need to think through the case when `queries` is used and there are multiple queries

Reply:
I don't think it needs to be on metadata - just my misunderstanding of how to make that task data available. Can remove from metadata and just parse the query types/args into a struct where we need them.